ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -60,6 +101,23 @@ const SnapUIRendererComponent = ({
     [content],
   );
 
+  const footerSections = useMemo(
+    () =>
+      footer &&
+      mapToTemplate({
+        map: {},
+        element: footer,
+        footer: true,
+      }),
+    [footer],
+  );
+
+  const defaultButtons = footer && getDefaultButtons(footer, onCancel);
```

Comment:
This seems complicated, wondering if this should simply happen in the footer mapper

Reply:
We could do this, I just wanted to avoid passing another thing to `mapToTemplate` since I need to pass `onCancel` to the buttons.

Reply:
Yeah, thats not ideal either 🤔